### PR TITLE
Issue #1485 - Intro acknowledgement page added to domain request form for new requests only

### LINKED ISSUE #1485: Inform users of .gov requirements before starting a new domain request
### Issue description

This ticket is for implementing the designs for https://github.com/cisagov/manage.get.gov/issues/547, also known as the preamble.

Flow from **beta.get.gov**:
- a user clicks start a new domain requests, moves past a modal and then logs in
- they arrive at a new application (in manage.get.gov) that has  a new page is inserted before the "org type" step

This ticket is for just adding the new page and having it show conditionally based on the user arriving from beta.get.govs links/buttons

### Acceptance criteria

- [ ] The request flow displays a new, informational page (i.e. preamble) at the beginning of the form when a user arrived at the application not via the edit an existing application button (OR link on beta.get.gov see related ticket)

### Additional context

- [New text for both flows](https://docs.google.com/document/d/15FpRLKXGeeNJ5D8uCu1YtRDZSWN5ZrJPKXJV4tCY6fw/edit#heading=h.cldogb82du8p)
- [Mural with flows](https://app.mural.co/t/gsa6/m/gsa6/1701893071504/3b10ecf79684c8e3a1387d7fe5665b58f3c156fa?sender=michellerago0372)
- Figma mockup: [Flow from beta.get.gov](https://www.figma.com/proto/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?page-id=2913%3A27400&type=design&node-id=2927-42517&viewport=131%2C74%2C0.14&t=l70YtUOmnoE59X5H-1&scaling=min-zoom&starting-point-node-id=2927%3A42517&show-proto-sidebar=1&mode=design) and [flow from manage.get.gov](https://www.figma.com/proto/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?page-id=2913%3A27400&type=design&node-id=2926-28192&viewport=131%2C74%2C0.14&t=VIQ4EPkC3bwHn9Xk-1&scaling=min-zoom&starting-point-node-id=2926%3A28192&show-proto-sidebar=1&mode=design)

The related ticket needs to wait until launch to be merged, but this ticket can be merged without an issue.

Design: https://github.com/cisagov/manage.get.gov/issues/547
Related to cisagov/getgov-home#147

### Links to other issues

_No response_

### PR DESCRIPTION
**Pull request by dave-kennedy-ecs**
## Ticket

Resolves #1485 

## Changes

- Create a domain request button ("/register/" link) on home page now forces user to acknowledge an informative message before accessing the request form.
- Existing requests, which have previously been started, do not get this prompt. 

## Context for reviewers

On Home page, click 'Start a new domain request'.  When doing so, you should see the intro message.

On Home page, from list of domain requests, click 'Edit' next to a Started domain request.  When doing so, you should not see the intro message.  Instead, you should directly access the application form.

Developer notes for code review:
views/application.py: this is where the workflow changes are.  The changes are to the behavior of both get and post methods. The behavior is in keeping with the current implementation of the application form, which is very url-dependent.
test/test_views.py: added one simple test, and modified several which were expecting the first page of the application process to be the organization type page
templates/application_intro.html: this is the template for the intro page

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [x] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [x] Checked for landmarks, page heading structure, and links
- [x] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [x] Checked keyboard navigability
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [x] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/tests/test_views.py [hunk 1]
```diff
@@ -137,11 +137,31 @@ def setUp(self):
         self.app.set_user(self.user.username)
         self.TITLES = ApplicationWizard.TITLES
 
+    def test_application_form_intro_acknowledgement(self):
+        """Tests that user is presented with intro acknowledgement page"""
+        intro_page = self.app.get(reverse("application:"))
+        self.assertContains(intro_page, "You're about to start your .gov domain request")
+
```
Comment (rachidatecs) [inline src/registrar/tests/test_views.py:144]: Wrote this test asserting that when you access an application through clicking 'Edit', you bypass the intro page. It works:

```
def test_application_form_intro_is_skipped_when_edit_access(self):
        """Tests that user is NOT presented with intro acknowledgement page when accessed through 'edit'"""        
        completed_application(status=DomainApplication.ApplicationStatus.STARTED, user=self.user)
        home_page = self.app.get("/")
        self.assertContains(home_page, "city.gov")
        # click the "Edit" link
        detail_page = home_page.click("Edit", index=0)
        # Check that the response is a redirect
        self.assertEqual(detail_page.status_code, 302)
        # You can access the 'Location' header to get the redirect URL
        redirect_url = detail_page.url
        self.assertEqual(redirect_url, "/register/organization_type/")
```

**Review by rachidatecs — COMMENTED** [review]
**Before merging:** See additional test asserting a bypass of the intro.

File: src/registrar/tests/test_views.py (hunk 1 above)
Comment (dave-kennedy-ecs) [reply src/registrar/tests/test_views.py:144]: Thanks.  I added the test

**Review by rachidatecs — APPROVED** [review]

**Review by therealslimhsiehdy — APPROVED** [review]
Screenshot 2023-12-29 at 10 46 41 AM

LGTM! Was able to see the new info page for a new domain request, and not see that info page for any of the already started or already created domain requests -- tests also look great.

File: src/registrar/templates/application_intro.html [hunk 2]
```diff
@@ -10,11 +10,11 @@
 
         <h1>You're about to start your .gov domain request.</h1>
```
Comment (gabydisarli) [inline src/registrar/templates/application_intro.html:11]: @michelle-rago can we change all instances of ' to ‘ ?

Comment (dave-kennedy-ecs) [reply src/registrar/templates/application_intro.html]: ok, done.

**Review by gabydisarli — APPROVED** [review]
Looks great!

**Pull request merged**